client/src/quic_client.rs

```diff
@@ -312,3 +312,88 @@ impl QuicClient {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use {
+        crate::{
+            quic_client::QuicTpuConnection,
+            tpu_connection::{ClientStats, TpuConnection},
+        },
+        crossbeam_channel::unbounded,
+        solana_sdk::{packet::PACKET_DATA_SIZE, quic::QUIC_PORT_OFFSET, signature::Keypair},
+        solana_streamer::quic::spawn_server,
+        std::{
+            collections::HashMap,
+            net::{SocketAddr, UdpSocket},
+            sync::{
+                atomic::{AtomicBool, Ordering},
+                Arc, RwLock,
+            },
+            time::{Duration, Instant},
+        },
+    };
+
+    #[test]
+    fn test_quic_client_multiple_writes() {
+        solana_logger::setup();
+        let s = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let exit = Arc::new(AtomicBool::new(false));
+        let (sender, receiver) = unbounded();
+        let keypair = Keypair::new();
+        let ip = "127.0.0.1".parse().unwrap();
+        let staked_nodes = Arc::new(RwLock::new(HashMap::new()));
+        let t = spawn_server(
+            s.try_clone().unwrap(),
+            &keypair,
+            ip,
+            sender,
+            exit.clone(),
+            1,
+            staked_nodes,
+            10,
+            10,
+        )
+        .unwrap();
+
+        let addr = s.local_addr().unwrap().ip();
+        let port = s.local_addr().unwrap().port() - QUIC_PORT_OFFSET;
+        let tpu_addr = SocketAddr::new(addr, port);
+        let client = QuicTpuConnection::new(UdpSocket::bind("127.0.0.1:0").unwrap(), tpu_addr);
+
+        // Send a full size packet with single byte writes.
+        let num_bytes = PACKET_DATA_SIZE;
+        let num_expected_packets: usize = 10000;
+        let packets: Vec<_> = (0..num_expected_packets)
+            .map(|_idx| -> Vec<u8> { [0; PACKET_DATA_SIZE].to_vec() })
+            .collect();
+
+        let stats = Arc::new(ClientStats::default());
+
+        assert!(client
+            .send_wire_transaction_batch_async(packets, stats)
+            .is_ok());
+
+        let mut all_packets = vec![];
+        let now = Instant::now();
+        let mut total_packets = 0;
+        while now.elapsed().as_secs() < 15 {
```

Comment:
how long does this test take to run?

Reply:
On my local machine, I actually see it succeeding in 5 seconds for 10000 packets, but I saw a failure on CI with the timeout, so I increased it. I did later reduce the timeout and timeout. I suspect that some of this is due to the latency from connecting for the first time, and it's notable that in the quic server tests, which this test is based on, there was a 5 second timeout on sending a single packet.